great_expectations/data_context/data_context/data_context.py

```diff
@@ -24,6 +26,43 @@
     from great_expectations.alias_types import PathStr
 
 
+@overload
```

Comment:
For each overload, should we add a docstring or comment to describe what situation they're used in? I know the answer here since I worked on it but it might be helpful for others?

Reply:
That sounds like a good idea.
I probably won't have the best explanation but I'll try.

Reply:
@cdkini Done